services/venice-controller/src/main/java/com/linkedin/venice/controller/stats/SparkServerStats.java

```diff
@@ -1,57 +1,198 @@
 package com.linkedin.venice.controller.stats;
 
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.HTTP_REQUEST_METHOD;
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.HTTP_REQUEST_URI;
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.HTTP_RESPONSE_STATUS_CODE;
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.HTTP_RESPONSE_STATUS_CODE_CATEGORY;
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.VENICE_CLUSTER_NAME;
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.VENICE_RESPONSE_STATUS_CODE_CATEGORY;
+
+import com.google.common.collect.ImmutableMap;
+import com.linkedin.venice.controllerapi.ControllerRoute;
 import com.linkedin.venice.stats.AbstractVeniceStats;
-import com.linkedin.venice.stats.TehutiUtils;
+import com.linkedin.venice.stats.VeniceMetricsConfig;
+import com.linkedin.venice.stats.VeniceMetricsRepository;
+import com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository;
+import com.linkedin.venice.stats.dimensions.HttpResponseStatusCodeCategory;
+import com.linkedin.venice.stats.dimensions.HttpResponseStatusEnum;
+import com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions;
+import com.linkedin.venice.stats.dimensions.VeniceResponseStatusCategory;
+import com.linkedin.venice.stats.metrics.MetricEntityStateGeneric;
+import com.linkedin.venice.stats.metrics.TehutiMetricNameEnum;
 import io.tehuti.metrics.MetricsRepository;
 import io.tehuti.metrics.Sensor;
 import io.tehuti.metrics.stats.Count;
 import io.tehuti.metrics.stats.OccurrenceRate;
-import io.tehuti.metrics.stats.Total;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import spark.Request;
+import spark.Response;
 
 
 public class SparkServerStats extends AbstractVeniceStats {
-  final private Sensor requests;
-  final private Sensor finishedRequests;
-  final private Sensor successfulRequest;
-  final private Sensor failedRequest;
-  final private Sensor successfulRequestLatency;
-  final private Sensor failedRequestLatency;
-  final private Sensor currentInFlightRequestTotal;
+  private final boolean emitOpenTelemetryMetrics;
+  private final VeniceOpenTelemetryMetricsRepository otelRepository;
+  private final Map<VeniceMetricsDimensions, String> baseDimensionsMap;
+
+  private final Sensor requests;
+  private final Sensor finishedRequests;
+
+  private final MetricEntityStateGeneric successfulRequestCountMetric;
```

Comment:
Lets not use `MetricEntityStateGeneric` unless there is a string as a dimension that is not part of the base dimensions. In this cases, leaving out cluster name, there are 5 enums.

Raised a PR https://github.com/linkedin/venice/pull/2159 to support that.